app/models/lettings_log.rb

```diff
@@ -79,6 +80,10 @@ def form_name
     FormHandler.instance.form_name_from_start_year(collection_start_year, "lettings")
   end
 
+  def clear_invalid_radio_answers!
+    super if startdate_was && @start_year != FormHandler.instance.collection_start_year(startdate_was)
```

Comment:
you might have tried this already, but sales logs have a startdate method that just sets startdate = saledate, so you might be able to put all this logic on the log model without separating out for lettings and sales

Reply:
didn't know about attribute_was though, cool!

Reply:
I forgot this, but given that the attribute_was thing is coming from rails magic I think it would be a bit weird to make a parallel method for just this situation.
```
def startdate_was
  saledate_was
end
```

Reply:
yeah probably fair enough 👍